### PR TITLE
Adding support of basic authentication and support to import multiple certificates to truststore 

### PR DESCRIPTION
**Pull request by Shehanir**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: apim-operator/build/Dockerfile [hunk 1]
```diff
@@ -10,7 +10,8 @@ COPY build/_output/bin/apim-operator ${OPERATOR}
 COPY build/microgwconf.mustache /usr/local/bin
 COPY build/controller_resources/Dockerfile /usr/local/bin
 COPY build/dockerSecretTemplate.mustache /usr/local/bin
-COPY build/controller_resources/dockerFile.mustache /usr/local/bin
+#COPY build/controller_resources/dockerFile.mustache /usr/local/bin
```
Comment (pubudu538) [inline apim-operator/build/Dockerfile]: If we are going with gotmpl, then let's remove this line. No need to add commented code.

File: apim-operator/build/controller_resources/dockerFile.gotmpl [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#This image will be used by Kaniko pod to create the micro-gw image
```
Comment (pubudu538) [inline apim-operator/build/controller_resources/dockerFile.gotmpl]: Let's add the license header.. For all the files except gen files, please add the license header

File: apim-operator/build/Dockerfile (hunk 1 above)
Comment (Shehanir) [reply apim-operator/build/Dockerfile]: removed

File: apim-operator/build/controller_resources/dockerFile.gotmpl (hunk 2 above)
Comment (Shehanir) [reply apim-operator/build/controller_resources/dockerFile.gotmpl]: added

File: apim-operator/pkg/controller/api/api_controller.go [hunk 3]
```diff
@@ -226,29 +250,55 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+	jobVolumeMount, jobVolume := getVolumes(instance)
 
-	if errc != nil && errors.IsNotFound(errc) {
-		reqLogger.Info("defined cretificate is not found")
-		return reconcile.Result{}, errc
-	}
+	//get certificate for JWT and Oauth
+	if strings.EqualFold(security.Spec.Type, "Oauth") || strings.EqualFold(security.Spec.Type, "JWT") {
+
+		errc := r.client.Get(context.TODO(), types.NamespacedName{Name: security.Spec.Certificate, Namespace: wso2NameSpaceConst}, certificateSecret)
```
Comment (vithu30) [inline apim-operator/pkg/controller/api/api_controller.go]: The line length is exceeded. Shall we split this into new line?

File: apim-operator/pkg/controller/api/api_controller.go [hunk 4]
```diff
@@ -595,19 +658,28 @@ func getCredentials(r *ReconcileAPI, name string) error {
...
 		if strings.EqualFold(k, "username") {
-			basicUsername = string(v)
+			usrname = string(v)
 		}
 		if strings.EqualFold(k, "password") {
-			//encode password to sha1
-			_, err := hasher.Write([]byte(v))
+			password = v
+		}
+
```
Comment (vithu30) [inline apim-operator/pkg/controller/api/api_controller.go]: Shall we remove the unnecessary new lines?

Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go:666]: This is already fixed in the latest code

File: apim-operator/pkg/controller/api/api_controller.go (hunk 3 above)
Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go:266]: Resolved

**Pull request merged**